src/ansys/fluent/core/solver/error_message.py

```diff
@@ -25,3 +25,55 @@ def allowed_values_error(
     context: str, trial_name: str, allowed_values: List[str]
 ) -> ValueError:
     return ValueError(allowed_name_error_message(context, trial_name, allowed_values))
+
+
+class FluentConnectionError(ValueError):
+    """Custom Fluent connection errors."""
+
+    class PortNotProvidedError(ValueError):
+        def __init__(self, error):
+            super().__init__(error)
+
+    class StartTimeoutError(RuntimeError):
+        def __init__(self, error):
+            super().__init__(error)
+
+    class RemoteNotSupportedError(ValueError):
+        def __init__(self, error):
+            super().__init__(error)
+
+    class WaitTypeError(TypeError):
+        def __init__(self, error):
+            super().__init__(error)
+
+
+class LauncherError(RuntimeError):
+    """Custom Fluent launch errors."""
+
+    class MeshingModeError(RuntimeError):
+        def __init__(self, error):
+            super().__init__(error)
+
+    class UnexpectedKeywordArgumentError(TypeError):
+        def __init__(self, error):
+            super().__init__(error)
+
+    class DockerContainerLaunchNotSupportedError(SystemError):
+        def __init__(self, error):
+            super().__init__(error)
+
+
+class TransactionError(RuntimeError):
+    """Custom transaction errors."""
+
+    class SurfaceNamesIDsNotProvidedError(RuntimeError):
```

Comment:
Something not provided sounds like a `TypeError`. CHeck if the new type is really needed. If you focus on testing then we can see what is truly needed.

Reply:
Yes, there are many places where incorrect exceptions has been raised. I'm not focusing on correcting each of them right now, I will first ensure that all of them have been covered as per new exception format then after passing all tests, I will correct those exceptions one by one.

Reply:
@seanpearsonuk This should be `ValueError` not `TypeError`.

"`TypeError` is raised when an operation or function is applied to an object of inappropriate type. The associated value is a string giving details about the type mismatch."

Reply:
Corrected error type.